### PR TITLE
Adding back NDF dataset support with some feature additions

### PR DESCRIPTION
**Pull request by oadonca**

### CODE REVIEW

File: configs/dataset/ndf/_default.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 
 # This is the general structure of the datamodule for NDF tasks.
+defaults:
+  - ../@train_dset.demo_dset.occlusion_cfg: _augmentations
```
Comment (beneisner) [inline configs/dataset/ndf/_default.yaml:4]: lol this is unholy. love it.

File: taxpose/datasets/enums.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from enum import Enum
```
Comment (beneisner) [inline taxpose/datasets/enums.py:1]: this is fine - but I don't really like the ObjectClass / phase stuff anymore. doesn't really match my new mental model of things.

Comment (beneisner) [reply taxpose/datasets/enums.py:1]: fine for now tho

File: taxpose/datasets/ndf.py [hunk 3]
```diff
@@ -78,6 +74,13 @@ class NDFPointCloudDatasetConfig:
     action: Phase = Phase.GRASP
     symmetry_after_transform: bool = False
 
+    # Augmentation parameters.
+    occlusion_cfg: Optional[OcclusionConfig] = None
+    synthetic_occlusion: bool = False
```
Comment (beneisner) [inline taxpose/datasets/ndf.py:79]: nit: cant you just check for existence of occlusion_cfg?

**Review by beneisner — APPROVED** [review]
Looks good - let me know if it trains lol.

File: configs/dataset/ndf/_default.yaml (hunk 1 above)
Comment (oadonca) [reply configs/dataset/ndf/_default.yaml:4]: yeah probably a better way to do it, but it works :)

File: taxpose/datasets/enums.py (hunk 2 above)
Comment (oadonca) [reply taxpose/datasets/enums.py:1]: I see, mostly did this shifting around to prevent a circular import

File: taxpose/datasets/ndf.py (hunk 3 above)
Comment (oadonca) [reply taxpose/datasets/ndf.py]: added

**Pull request merged**